File: C#1/Dto_Command/Buy.cs
using System;
using System.Collections.Generic;

namespace DTO_Command;

public partial class Buy
{
    public short Id { get; set; }

    public string? CodeClient { get; set; }

    public DateTime? Date { get; set; }

    public double? SumPrice { get; set; }

    public string? Note { get; set; }

    public List<Product>? products { get; set; }
    public List<Product>? finished { get; set; }
}

File: patio_furniture/src/app/classes/Buy.ts
import { Product } from "./Product";
import { PurchaseBuy } from "./PurchaseBuy";

export class Buy {
    constructor(
        public id?: number,
        public codeClient?: string,
        public date?: Date,
        public sumPrice: number=0,
        public note?: string,
        public products?: Array<Product>,
        public finished:Array<Product>=new Array<Product>(),
        public count:number=0
    ) { }
}